model/commitqueue/commit_queue.go

```diff
@@ -111,6 +109,34 @@ func (q *CommitQueue) Next() (CommitQueueItem, bool) {
 	return q.Queue[0], true
 }
 
+func (q *CommitQueue) NextUnprocessed(n int) []CommitQueueItem {
+	items := []CommitQueueItem{}
+	for i, item := range q.Queue {
+		if item.Version != "" {
+			continue
+		}
+		if i+1 > n {
```

Comment:
can we do this check before 115 so we don't keep continuing before the short circuit? (I _know_ the queue is never long enough for this to really matter but)

Reply:
yep changed